### PR TITLE
Add javadocs and a little cleanup

### PR DESCRIPTION
**Pull request by Coffee-Boyy**
I added the javadocs to the inherited methods from Swirlds as well as fixed the SDK download link in the README and added a gitignore for intellij and eclipse!

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,24 +4,22 @@
...
     <artifactId>HelloSwirld</artifactId>
     <version>1.0-SNAPSHOT</version>
 
     <properties>
-        <sdk.dir>/home/swirlds/sdk</sdk.dir>
-        <app.name>HelloSwirld</app.name>
-
-        <app.repo>${sdk.dir}/data/repos/app/</app.repo>
-        <platform.repo>${sdk.dir}/data/repos/platform/</platform.repo>
+        <swirlds.sdk>/users/connor/Documents/hashgraph/sdk</swirlds.sdk>
```
Comment (dappcoder) [inline pom.xml]: SDK location probably leaked from your local setup. I propose some generic unix path (or revert to "/home/swirlds/sdk") if nothing else comes to mind.

File: README.md [hunk 2]
```diff
@@ -1,47 +1,49 @@
...
 
-It allows you to build and run hashgraph apps using the SDK more like an external resource rather than creating new modules along the demo apps inside the SDK directory structure.
+It allows you to build and run Hashgraph apps using the SDK more like an external resource rather than creating new modules along the demo apps inside the SDK directory structure.
 
-Clone this repo to whatever location (independend from the Hashgraph sdk).
-#### Install Hashgraph SDK
-Follow the pre-requisites, download and unzip the SDK to preferred location. See https://dev.hashgraph.com/docs/installation/
+Clone this repo to whatever location (independent from the Hashgraph sdk).
+## Install Hashgraph SDK
+Follow the pre-requisites, download and unzip the SDK to preferred location. See https://www.swirlds.com/download/
```
Comment (dappcoder) [inline README.md]: I think the previous url (https://dev.hashgraph.com/docs/installation/) is a better fit as it describes the prerequisites for installing the Hashgraph SDK (e.g. JCE) which are not reiterated here. That page also includes the download location for the sdk itself.

**Review by dappcoder — CHANGES_REQUESTED** [review]
Hi Connor!
Thanks so much for your contribution. I appreciate the cleanup. Please see a couple of comments on your changes.
I think this can evolve into some kind of a maven archetype or another template that would automatically allow developers to create apps from scratch.
Nice to meet you! 
Alex.

File: pom.xml [hunk 3]
```diff
@@ -81,19 +87,19 @@
...
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-jar-plugin</artifactId>
-                <version>2.3.2</version>
+                <version>3.1.0</version>
                 <configuration>
                     <archive>
                         <manifest>
-                            <mainClass>${app.name}Main</mainClass>
+                            <mainClass>Main</mainClass>
```
Comment (dappcoder) [inline pom.xml]: I've built and run the module with HelloSwirldMain renamed to Main (and java classes renamed accordingly). It seems to be stuck and the console windows do not open. The SDK uses that prefix as an identifier for the app, so can't do without it.

**Comment by Coffee-Boyy** [comment]
Comments have been addressed

I was also thinking about how convenient it would be if this was converted into an archetype! Nonetheless, thanks for the starter template.

This hashgraph technology is very interesting and I wish to delve deeper into the underlying technologies. I am quite curious about their centralized governance structure tho, not sure if that will have impact on addoption

**Review by dappcoder — CHANGES_REQUESTED** [review]
<mainClass>Main</mainClass> freezes Swirlds Browser. It expects a prefix before "Main" otherwise, it won't run. 
I suggest reverting the mainClass prop to HelloSwirldMain, the Main java class name to HelloSwirldMain and the State java class to HelloSwirldState.

**Comment by Coffee-Boyy** [comment]
Good catch! I also added a note to the readme so nobody else runs into the same issue.

**Review by dappcoder — APPROVED** [review]

**Pull request merged**